source/core/src/core/gold/infrastructure/repositories/gold_measurements_repository.py

```diff
@@ -2,55 +2,59 @@
 
 from pyspark.sql import DataFrame
 
-import core.utility.shared_helpers as shared_helpers
+import core.gold.infrastructure.config.spark as spark_session
+import core.utility.delta_table_helper as delta_table_helper
 from core.gold.domain.constants.column_names.gold_measurements_column_names import GoldMeasurementsColumnNames
 from core.gold.infrastructure.config import GoldTableNames
+from core.settings import StorageAccountSettings
 from core.settings.gold_settings import GoldSettings
-from core.utility.environment_variable_helper import EnvironmentVariable, get_env_variable_or_throw
+from core.settings.streaming_settings import StreamingSettings
+from core.utility import shared_helpers
 
 
 class GoldMeasurementsRepository:
     def __init__(self) -> None:
         self.gold_container_name = GoldSettings().gold_container_name
         self.gold_database_name = GoldSettings().gold_database_name
+        self.data_lake_settings = StorageAccountSettings().DATALAKE_STORAGE_ACCOUNT
         self.table = f"{self.gold_database_name}.{GoldTableNames.gold_measurements}"
 
-    def start_write_stream(
+    def write_stream(
         self,
-        df_source_stream: DataFrame,
+        source_table: DataFrame,
         batch_operation: Callable[["DataFrame", int], None],
-        terminate_on_empty: bool = False,
-    ) -> None:
+    ) -> bool | None:
         query_name = "measurements_silver_to_gold"
-        datalake_storage_account = get_env_variable_or_throw(EnvironmentVariable.DATALAKE_STORAGE_ACCOUNT)
         checkpoint_location = shared_helpers.get_checkpoint_path(
-            datalake_storage_account, self.gold_container_name, GoldTableNames.gold_measurements
+            self.data_lake_settings, self.gold_container_name, GoldTableNames.gold_measurements
         )
-        df_write_stream = (
-            df_source_stream.writeStream.format("delta")
+
+        write_stream = (
+            source_table.writeStream.format("delta")
             .queryName(query_name)
             .option("checkpointLocation", checkpoint_location)
-            .foreachBatch(batch_operation)
         )
 
-        if terminate_on_empty:
-            df_write_stream.trigger(availableNow=True).start().awaitTermination()
-        else:
-            df_write_stream.start().awaitTermination()
+        stream_settings = StreamingSettings()
+
+        if stream_settings.continuous_streaming_enabled is False:
```

Comment:
```suggestion
        if StreamingSettings().continuous_streaming_enabled is False:
```
I think this would be cleaner, since we're not using the `stream_settings` in other places beneath